app/Models/User.php

```diff
@@ -1339,21 +1339,24 @@ public function statisticsTaiko()
         return $this->hasOne(UserStatistics\Taiko::class);
     }
 
-    public function statistics(string $ruleset, bool $returnQuery = false, ?string $variant = null)
+    public static function statisticsRelationName(string $ruleset, ?string $variant = null): ?string
```

Comment:
static functions should be at the top (this class is already relatively messy but let's not make it worse)